runtime/controller/connection.go

```diff
@@ -44,23 +49,23 @@ func (o *ConnectionOptions) BindFlags(fs *pflag.FlagSet) {
 		"Allow the controller to make HTTP requests to external services like insecure Git servers, container registries, etc.")
 }
 
-// CheckEnvironmentCompatibility checks if the enviornment is compatible with
+// CheckEnvironmentCompatibility checks if the environment is compatible with
 // the configured connection options.
 func (o *ConnectionOptions) CheckEnvironmentCompatibility() error {
 	if !o.AllowHTTP {
 		config := httpproxy.FromEnvironment()
 		if config.HTTPProxy != "" {
-			return fmt.Errorf("usage of HTTP requests is blocked but found a HTTP proxy set in enviornment")
+			return fmt.Errorf("%w: found HTTP proxy set in environment", ErrInsecureHTTPBlocked)
```

Comment:
the reason is that its very hard to test environment proxy configurations in Golang, since the `httpproxy.FromEnvironment()` caches the value of `HTTP_PROXY` and `HTTPS_PROXY`, so we can't manipulate the values for the various test cases that might be needed. If we want to write tests that cover both the http and https conditions in a reliable way, we would need to do one of the following:
* remove the usage of `httpproxy.FromEnvironment()` and read the env vars ourselves using os.Getenv()
* override the proxy config variable using dependency injection

i'm more inclined to the first option. this function is supposed to be call at controller startup, so there's no real advantage for us anyway if the values are cached.

Reply:
added a test using a private global variable